### DISCUSSION TITLE
Progressive Web Apps

### DISCUSSION THREAD

**Discussion by wbonekamp** [oqtane/oqtane.framework#1844] [Ideas]
What are the options to create Progressive Web Apps with Oqtane?

Maintaining native apps or apps based on Xamarin takes more and more time...
Do you think PWA is the future?

**Comment by sbwalker** [comment]
@wbonekamp Oqtane already supports PWAs - both when running on Blazor Server and Blazor WebAssembly. You can enable it in the Site Settings of any site:

![image](https://user-images.githubusercontent.com/4840590/144506321-d423e7a2-f16d-43c5-90b2-7e8481fc23a2.png)

Enabling these options will automatically inject the service worker and manifest into the page.
